clang/utils/generate_unsupported_in_drivermode.py

```diff
@@ -0,0 +1,534 @@
+    check_str: The string or regex to be sent to FileCheck
+    supported_joined_option_sequence: List of DriverOption objects for supported options
+                                      that are Kind *JOINED*, as defined in Options.td
+    supported_non_joined_option_sequence: List of DriverOption objects for supported options
+                                          that are not Kind *JOINED*, as defined in Options.td
+    test_option_sequence: A list of all the prefix-option pairs that will be tested for this driver
+    """
+
+    lit_cmd_prefix: str
+    lit_cmd_options: str
+    visibility_str: str
+    lit_cmd_end: str
+    check_str: str
+    supported_joined_option_sequence: list[DriverOption] = dataclasses.field(
+        default_factory=list
+    )
+    supported_non_joined_option_sequence: list[DriverOption] = dataclasses.field(
+        default_factory=list
+    )
+    test_option_sequence: list[str] = dataclasses.field(default_factory=list)
+
+
+def collect_transitive_groups(member, options_dictionary):
+    """Find the groups for a given member, where a member can be an option or a group.
+    Note that groups can themselves be part of groups, hence the recursion
+
+    For example, considering option 'C', it has the following 'Group' field as defined by Options.td:
+      "C": {
+        "Group": {
+          "def": "Preprocessor_Group",
+          // ...
+        },
+        // ...
+      },
+    'Preprocessor_Group' is itself part of 'CompileOnly_Group', so option 'C' would be part of both groups
+      "Preprocessor_Group": {
+        // ...
+        "Group": {
+          "def": "CompileOnly_Group",
+          // ...
+        },
+        // ...
+      },
+
+    member: An option object or group object from Options.td.
+    options_dictionary: The converted Python dictionary from the Options.td json string
+
+    Return: A set including the group(s) found for the member. If no groups found, returns an empty set
+    """
+    parent_field = options_dictionary[member]["Group"]
+    if parent_field is None:
+        return set()
+
+    parent_name = parent_field["def"]
+    return {parent_name} | collect_transitive_groups(parent_name, options_dictionary)
+
+
+def get_visibility(option):
+    """Get a list of drivers that a given option is exposed to
+    option: The option object from Options.td
+    Return: Set that contains the visibilities of the given option
+    """
+    visibility_set = set()
+    # Check for the option's explicit visibility
+    for visibility in options_dictionary[option]["Visibility"]:
+        if visibility is not None:
+            visibility_set.add(visibility["def"])
+
+    # Check for the option's group's visibility
+    group_set = collect_transitive_groups(option, options_dictionary)
+    if group_set:
+        for group_name in group_set:
+            for visibility in options_dictionary[group_name]["Visibility"]:
+                visibility_set.add(visibility["def"])
```

Comment:
```suggestion
     for group_name in collect_transitive_groups(option, options_dictionary):
         for visibility in options_dictionary[group_name]["Visibility"]:
             visibility_set.add(visibility["def"])
```
`collect_transitive_groups` shouldn't ever return `None` now.